### DISCUSSION TITLE
Why does the getAccountCoinAmount method in the SDK route to different URLs based on network type instead of using the configured private node URL?

### DISCUSSION THREAD

**Discussion by liucan163** [aptos-labs/aptos-developer-discussions#116] [Questions] [answered]
When I use the getAccountCoinAmount method, I noticed that in the SDK it actually goes to different official routes based on the configured network type. For example, when env: 'test', it defaults to https://indexer-testnet.staging.gcp.aptosdev.com/v1/graphql, but it does not use the private node URL I configured. Why is this?

**Comment by 0xmaayan** [comment, marked as answer]
We use fullnode and/or indexer to fetch data from the Aptos chain. Fullnode can be used to query current state data and Indexer can be used to query historical data or complex data we simply can not query directly from fullnode.

The Aptos TS SDK supports all Aptos RCPs, (fullnode, faucet and indexer), and resolves the query to the correct route. 
We could configure a custom server for the SDK to use when querying 

```
const aptosConfig = new AptosConfig({
  fullnode:"my-custom-fullnode",
  indexer:"my-custom-indexer"
})
```
That way, when the SDK uses fullnode or indexer to query data, it would hit the custom configured server.


As for the `getAccountCoinAmount` query, the SDK resolves it to use Aptos `Indexer` server.
If you want to use fullnode to query an account coin balance, you could query it like that
```
const resource = await aptos.getAccountResource({
  accountAddress: address,
  resourceType: COIN_TYPE, // for example 0x1::coin::CoinStore<0x1::aptos_coin::AptosCoin>
});
const amount = Number(resource.coin.value);
```

**Reply by liucan163** [reply]
ok, thank you very much!
